include/powsybl/stdcxx/instanceof.hpp

```diff
@@ -22,6 +24,16 @@ inline bool isInstanceOf(const std::unique_ptr<T>& pointer) {
     return dynamic_cast<Base*>(pointer.get()) != nullptr;
 }
 
+template <typename Base, typename T>
+inline bool isInstanceOf(const stdcxx::CReference<T>& object) {
+    return static_cast<bool>(object) && std::is_same<Base, T>::value ? true : dynamic_cast<const Base*>(&object.get()) != nullptr;
```

Comment:
why don't you call the `isInstance(const T&)` functions instead duplicate the code?